model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java

```diff
@@ -265,9 +265,9 @@ public SlotHandle getScratchPad() {
 	 * @return the iterator of included scripts.
 	 */
 
-	public Iterator includeLibraryScriptsIterator() {
-		List libList = module.getAllLibraries();
-		List includeLibScriptList = new ArrayList();
+	public Iterator<Object> includeLibraryScriptsIterator() {
```

Comment:
This doesn't compile, but the Javadoc says the element type IncludeScriptHandle so if you're going to parameterize it, better confirm to the expectations set by the Javadoc:
```
	public Iterator<IncludeScriptHandle> includeLibraryScriptsIterator() {
		List<Library> libList = module.getAllLibraries();
		List<Object> includeLibScriptList = new ArrayList<Object>();
		if (libList != null) {
			for (Library lib : libList) {
				PropertyHandle propHandle = lib.getHandle(lib).getPropertyHandle(INCLUDE_SCRIPTS_PROP);

				Iterator scriptIter = propHandle.iterator();
				while (scriptIter.hasNext()) {
					includeLibScriptList.add(scriptIter.next());
				}
			}
		}
		return includeLibScriptList.iterator();
	}
```

Reply:
I will take a look into it.

Reply:
I will remove the type from the Iterator
because it causes a converting issue, but I will add your suggestion for the better loop.

![grafik](https://github.com/user-attachments/assets/daf00cdc-8541-4120-9b7c-e46fc90b2d31)

Reply:
I would do an instance of test in the loop. 

My primary concern is that once you replace the raw type with a parameterized type it becomes very hard to correct it in the future.

Reply:
Loop is changed, raw type without parameter (like original version).